test/windowing.jl

```diff
@@ -0,0 +1,15 @@
+using TransitionIndicators, Test, StatsBase
+
+@testset "sliding variance over vector" begin
+    n = 100
+    x = collect(1:n)
+    halfwidth = Int(ceil(rand() * (n/2-1)))     # Draw integer on [1, n/2-1]. Any of those should work.
+    stride = 2
+    ground_truth = var(1:2*halfwidth+1)         # x is a linear function (x = t) --> variance independent of window.
+
+    wv = WindowViewer(x, halfwidth, stride)
+    variance_window_viewer = map(StatsBase.var, wv)
```

Comment:
Did you measure the performance of this? I would instruct to measure the performance of the in place version of map, that shoudln't allocate. I.e., measure:
```
wv = WindowViewer(...)
out = zeros(length(wv))
@btime map!($var, $out, $wv)
```
What's the result?

Reply:
Just tried out the in place version but `map!` does not accept a `WindowViewer`as input. Does anything needs to be changed in the redefinition of `Base.iterate` for this to work?

Reply:
please be specific about the error message. what does it say. `MethodError`?

Reply:
you don't have to use `map!` just write a for loop over the elements of the WindowViewer iterator.

Reply:
`map!` now work and is of course more efficient than `map`. Required generating a vector out of the iterator elements.

Reply:
but what is the result though? can you post the benchmark?